app/views/building_map/index.html.erb

```diff
@@ -9,3 +9,46 @@
     polylines: BuildingMapHelper.leaflet_polylines(@route),
     markers: BuildingMapHelper.leaflet_markers(@route, @target))
 %>
+<style>
+.indoorLabel{
+  background: rgba(255, 255, 255, 0);
+  border:0;
+  border-radius:0px;
+  box-shadow: 0 0px 0px;
+}
+</style>
+<script>
+  function onEachFeature(feature, layer) {
+    layer.bindTooltip(
+      feature.properties.name,
+      {
+          permanent: true,
+          direction:'center',
+          className: 'indoorLabel',
+      }
+    );
+  }
+
+  function loadGeoJsonFile(filename) {
+    fetch(filename)
+    .then(response => response.json())
+    .then(geojsonFeature => {
+      rooms = L.geoJSON(geojsonFeature, { onEachFeature: onEachFeature }).addTo(map);
+      rooms.eachLayer((layer)=>{
+        layer.getTooltip().setLatLng(layer.getBounds().getCenter());
+      });
+      onZoom()
+    })
+  }
+
+  loadGeoJsonFile('assets/lecture-hall-building.geojson')
+  function onZoom() {
+    const zoomLevel = map.getZoom()
+    const tooltips = document.getElementsByClassName('indoorLabel')
+    const visibility = zoomLevel == 19 /* nearest zoom */ ? 'visible' : 'hidden'
+    for (let tooltip of tooltips) {
+      tooltip.style.visibility = visibility
```

Comment:
Why do you change the css values directly? I think I prefer something like this:
`map.eachLayer(layer => {if(layer.getTooltip()) layer.closeTooltip()})`